domain/secret/state/state.go

```diff
@@ -510,7 +505,7 @@ GROUP BY sm.secret_id`
 		return errors.Trace(err)
 	}
 
-	existingResult, err := dbSecrets.toSecretMetadata(dbsecretOwners)
+	existingResult, err := dbSecrets.toCoreSecretMetadata(dbsecretOwners)
```

Comment:
This existingResult is only used inside this method. And the only attributes we need are:
- owner
- rotate policy
- uri
- latest revision
- latest revision checksum

We want to stop using core in state. Can we add a TODO to fix this.